### DISCUSSION TITLE
Not sure what appends path params to my URL

### DISCUSSION THREAD

**Discussion by TomGruner85** [TanStack/router#2234] [Q&A]
Here's my setup:
router 1 (basepath: "/")
/app
--/app/$ ==> mount either remote component a or b, depending on splat
__root ==> contains two Links. One to /app/a (to="/app/$" params={{ _splat: 'a' }}) and one to /app/b (to="/app/$" params={{ _splat: 'b' }})

router 2 (basepath: "/app/a")
--/app/a (index)
--/app/a/about (static)

router 3 (basepath: "/app/b")
--/app/b (index)
--/app/b/about (static)

Now my problem is:
1. deep link into /app/a
2. click the link in router 1 app to navigate to /app/b
2. a. URL changes from /app/a to /app/b
3. in /app/b, navigate to about
3. a. URL changes from /app/b to /app/b/about
4. Now click the link in router 1 app to navigate to /app/a
4. a. **URL changes from /app/b/about to /app/a/app/b/about**
4. b. Now I see the default not found component because /app/a/app/b/about doesn't exist
5. I click on the link  in router 1 app to navigate to /app/a again
5. a. The URL changes from /app/a/app/b/about to /app/a but I still get the not found component

Why is this happening and how can I configure the router to not do that? I tried playing around with stale times and gc times, but could not prevent this from happening.

**Comment by TomGruner85** [comment]
![router1](https://github.com/user-attachments/assets/481f5b1a-8cfb-4710-948d-150bd7358681)
![router2](https://github.com/user-attachments/assets/dfad2d0d-ac43-4f80-b059-aa1e5483ce9b)

**Reply by schiller-manuel** [reply]
please provide a minimal.complete example by forking one of the existing router examples on stackblitz

**Reply by TomGruner85** [reply]
Sorry, it took me a little bit to reproduce this outside of my work setting. Stackblitz didn't work for me, because I don't know how to setup multiple apps within the same project. So I created a git repo here:
https://github.com/TomGruner85/react-router

Steps to reproduce:
1. go to localhost:3001
2. click the link "Navigate to /app/a"
3. now click the link "Navigate to /app/b"
4. Observe the URL changes to /app/a/app/b when the links "to" option has "/app/b"
